### DISCUSSION TITLE
Support XMP metadata: Displaying, searching, filtering, editing, exporting

### DISCUSSION THREAD

**Discussion by fahrstuhl** [ente/ente#1910] [Enhancements]
[XMP](https://en.wikipedia.org/wiki/Extensible_Metadata_Platform) seems like a widespread metadata format for images and videos. My camera, a Sony DSC RX100VA, uses it to store image ratings from 0 to 5 stars. The format also supports descriptions, keywords/tags, persons and items and their image regions.

It would be cool for ente to support importing, displaying, searching, filtering, editing and exporting XMP metadata. This could help with interoperability between ente and other software, cameras and existing image collections.

**Comment by lovetheguitar** [comment]
Apple photos also creates `.xmp` files when exporting images which leads to

> Ente does not support these file formats yet

during import into ente.

**Comment by StefanOltmann** [comment]
Yes, please support that all.

- Title
- DateTimeOrigina
- GPS
- Rating (0 to 5)
- Keywords/tags (dc subject)
- Persons in image
- Face regions (XMP-rwg-rs face tags)

Embedded as well as XMP sidecar (#4046).

**Comment by ua741** [comment]
> For some files, I can find the keywords under the “Subject” (xmp) field but some instead show under “weightedFlatSubject (xmp)”. In either case, the keywords do not seem to be indexed by the Magic search.

^ Adding note for future whenever we will add support for tags/keywords.
Related https://github.com/ente-io/ente/discussions/543

**Comment by r-cheologist** [comment]
A concrete migration use case, for what it's worth on prioritization:

I'm working through migrating a large family photo/video library from digiKam into Ente, and this thread is
 exactly the blocker. Wanted to add specifics in case it helps scope the work, since digiKam's XMP output f
ollows well-established conventions rather than anything proprietary:

**1. Face/person regions** — digiKam (like Lightroom, Capture One, and most other DAM tools) writes confirm
ed face tags using the **MWG (Metadata Working Group) Regions schema**: `Xmp.mwg-rs.Regions` / `mwg-rs:RegionList`, namespace `http://www.metadataworkinggroup.com/schemas/regions/`, with each region carrying `mwg-rs:Type="Face"`, `mwg-rs:Name` (person name), and a normalized `mwg-rs:Area` (x, y, w, h, 0–1 scale). It's the same schema #4048 is asking about. Since this is a cross-vendor standard (not digiKam-specific), reading it once would unlock migration from digiKam, Lightroom, Capture One, and others simultaneously — and would let Ente's own People feature be *seeded* from existing work instead of forcing a full re-tag of a library that may already have tens of thousands of photos tagged.

**2. Keyword tags** — digiKam writes hierarchical keywords to `digiKam:TagsList`, mirrored into `dc:subject` and `lr:hierarchicalSubject` (the same fields Lightroom uses, which is presumably what the `Subject` vs `weightedFlatSubject` inconsistency mentioned above is running into). Given [#543](https://github.com/ente/ente/discussions/543) is already labeled "planned," it'd be great if the underlying tag *storage* work there could also expose an XMP-import path, rather than only supporting tags created natively in Ente.

**3. Ratings/color labels** — `Xmp.xmp.Rating` (0–5, also mirrored to `Exif.Image.Rating`) is a near-universal field across photo software. I haven't found any Ente support for this.

None of this needs write-back to be useful as a first step — even read-only import (map MWG regions → Ente People at upload time, map `dc:subject`/hierarchical keywords → tags once [#543]( https://github.com/ente/ente/discussions/543) ships, map `xmp:Rating` → some equivalent) would remove the single biggest blocker for anyone migrating from a metadata-rich tool. Happy to provide sample files/XMP snippets from a real digiKam export if that's useful for testing.

Related: [#4048](https://github.com/ente/ente/discussions/4048), [#543](https://github.com/ente/ente/discussions/543)

**Reply by Gord067** [reply]
This is an excellent suggestion.  Having Ente work with Digikam would be great.  Digikam has plugin support, so, it might be possible to create an Export to Ente plugin, there are already quite a few export plugins, that would arrange all of the metadata in a form compatible with Ente. Just a thought and as I know nothing about programing I will leave the details to you.

**Comment by henri-sumit** [comment]
@ua741 is there any feedback on this?   Many users will not want to invest in a photo management system that does not 'play along' with industry standards (as bad as they may be).   Basic EXIF and XMP metadata has to be supported before I will invest my time cataloguing photos in Ente. One lesson I have learnt (and I think Ente could leverage this) is that people are far more hesitant to invest in 'closed system, cloud only' platforms.

**Comment by Not-Sure-Whats-Going-On** [comment]
Echoing what others have said, it is frustrating to see that 28 months after this issue was opened, it is still unresolved.  I would love to have Ente host my photos with a Pro hosting plan but that's not going to happen until searching IPTC / xmp tags works. 

After having invested hundreds of hours using Digikam to add meaningful descriptive tags to thousands of photos taken over the last 25 years or so, I am not going to invest in a hosting solution that ignores useful metadata like a description of the subject of the photo but can tell me unimportant technical details like what the atmospheric pressure was at the time the photo was taken.

It's such a shame because nearly everything else about Ente Photos is great but new features are being added while the essential basics like this are just left hanging.

**Comment by henrivdm** [comment]
This should be a simple, top priority: there are many users who want to mimic their desktop/file-based cataloguing in a cloud service, BUT in a way that the data remains with the owner:  - ie file-based, transferrable and current photo-cataloguing metadata (EXIF/IPTC/XMP)
As soon as Ente supports (1) standardised, transferrable tagging (2) proper 2-way Synch, I will invest in moving/synchronising my full library into Ente, unfortunately, not  before that.
